### PR TITLE
Allow fetching for different arch.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Add `--arch` flag for `brew fetch` to allow testing ARM cask downloads in CI. Also add `ensure` before all `Homebrew::SimulateSystem.clear`s.

### CODE REVIEW

**Review by Rylan12 — APPROVED** [review]

File: Library/Homebrew/cmd/fetch.rb [hunk 1]
```diff
@@ -20,6 +20,10 @@ def self.fetch_args
         Download a bottle (if available) or source packages for <formula>e
         and binaries for <cask>s. For files, also print SHA-256 checksums.
       EOS
+      # This is needed for testing cask downloads on CI.
+      flag "--arch=",
+           description: "Download for the given arch.",
```
[the remark this answers is not in the document]
Comment (reitermarkus) [reply Library/Homebrew/cmd/fetch.rb:25]: Expanded in https://github.com/Homebrew/brew/pull/15063.

File: Library/Homebrew/cmd/fetch.rb [hunk 2]
```diff
@@ -20,6 +20,10 @@ def self.fetch_args
...
+           hidden:      true
```
[the remark this answers is not in the document]
Comment (MikeMcQuaid) [reply Library/Homebrew/cmd/fetch.rb:26]: @reitermarkus IMO if it works elsewhere: it's useful to have?

[the remark this answers is not in the document]
Comment (reitermarkus) [reply Library/Homebrew/cmd/fetch.rb:26]: I don't really see a use case for this outside of CI. If there is one, we can make it public.

[the remark this answers is not in the document]
Comment (yurikoles) [reply Library/Homebrew/cmd/fetch.rb:26]: It's definitely useful for HBC contributors and developers, I think that it should be at least documented.

**Pull request merged**